union.go

```diff
@@ -0,0 +1,67 @@
+package rwvfs
+
+import (
+	"fmt"
+	"io"
+	"os"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
```

Comment:
Also, is it okay/desired that there is special handling of .go files inherited from `"golang.org/x/tools/godoc/vfs.NameSpace`, since this is a wrapper around that?

See https://godoc.org/golang.org/x/tools/godoc/vfs#NameSpace.ReadDir and http://gotools.org/golang.org/x/tools/godoc/vfs#namespace-go-L318-L330.